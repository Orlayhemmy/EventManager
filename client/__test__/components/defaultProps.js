export const props = {
  userState: {
    user: {
      isAdmin: true
    },
    isAuth: true
  },
  location: {
    pathname: '/'
  },
  center: {
    status: '',
    message: ''
  }
};

export const center = {
  status: 201,
  message: ''
};

export const bookedEvent = [
  { id: 1, eventTitle: 'Pizza Party', bookedDate: [], isApproved: true },
  { id: 2, eventTitle: 'Wedding Ceremony', bookedDate: [], isApproved: true  },
  { id: 3, eventTitle: 'Seminar', bookedDate: [], isApproved: false  }
];

export const logout = jest.fn();
export const onChange = jest.fn();
