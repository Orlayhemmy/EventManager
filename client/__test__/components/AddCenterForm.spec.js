import React from 'react';
import Connected, { CenterForm } from '../../components/CenterDetails/Template/Form/AddCenterForm';
import { center, userState } from './defaultProps';

describe('display centers', () => {

  const props = {
    // addCenterValidation: jest.fn(() => {
    //   const isValid = true;
    //   return isValid;
    // }),
    path: '/',
    center: {
      error: ''
    },
    uploadImage: jest.fn()
  };
  const event = {
    preventDefault: jest.fn(),
    target: {
      id: 'centerName',
      value: 'Balmoral',
      files: [{ data: 'picture.jpg', type: 'image/jpeg' }],
      result: ''
    }
  };

  const wrapper = shallow(<CenterForm {...props} />);
  it('should render the component based', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('call the onchange function and set state', () => {
    const spy = sinon.spy(wrapper.instance(), 'onChange');
    wrapper.instance().onChange(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().centerName).toEqual('Balmoral');
  });
  it('return true when there is no error on isValid call', () => {
    wrapper.setState({
      location: 'Lekki',
      description: 'A center with emotions',
      facilities: 'car park',
      capacity: '500',
      image: 'picture.jpg',
      cost: '50000'
    });
    const spy = sinon.spy(wrapper.instance(), 'isValid');

    wrapper.instance().isValid(event);
    expect(spy.calledOnce).toBeTruthy();

    wrapper.setState({
      location: 'Lekki#',
      description: 'A center with emotions',
      facilities: 'car park',
      capacity: '500',
      image: 'picture.jpg',
      cost: '50000'
    });
    wrapper.instance().isValid(event);
    expect(spy.calledTwice).toBeTruthy();
    expect(wrapper.state().errors.location).toEqual('location can not include symbols except comma');
  });
  it('call the submit function', () => {
    const spy = sinon.spy(wrapper.instance(), 'onSubmit');
    wrapper.find('#add-event').simulate('click');
    wrapper.setState({
      location: 'Lekki',
      description: 'A center with emotions',
      facilities: 'car park',
      capacity: '500',
      image: 'picture.jpg',
      cost: '50000'
    });
    wrapper.instance().onSubmit(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().location).toEqual('Lekki');
    expect(wrapper.state().capacity).toEqual('500');
  });
  it('call showImage function', () => {
    const spy = sinon.spy(wrapper.instance(), 'showImage');
    wrapper.instance().showImage(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().image).toEqual('picture.jpg');
  });
});

describe('Connected Component', () => {
  it('should render the connected component with all props', () => {

    const store = mockStore({ auth: { user: {}, center } });
    const wrapper = shallow(<Connected store={store} />);
    expect(wrapper.length).toEqual(1);
  });
});
