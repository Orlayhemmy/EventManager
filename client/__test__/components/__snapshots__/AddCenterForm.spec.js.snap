// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenterForm component should render the component without error 1`] = `
<form
  id="add-center-form"
  onSubmit={[Function]}
>
  <ImageUpload
    path="/"
    showImage={[Function]}
    uploadedImage=""
  />
  <span
    className="help-block"
  />
  <span
    className="main-color"
  >
    Title
  </span>
  <TextField
    id="centerName"
    onChange={[Function]}
    placeholder="Center name"
    type="text"
    value=""
  />
  <span
    className="main-color"
  >
     Location
  </span>
  <TextField
    id="location"
    onChange={[Function]}
    placeholder="Center location"
    type="text"
    value=""
  />
  <span
    className="main-color"
  >
     Facilities
  </span>
  <p
    className="subtitle"
  >
    separate facilities with commas. Do not include spaces
  </p>
  <TextField
    id="facilities"
    onChange={[Function]}
    placeholder="Facilities in center"
    type="text"
    value=""
  />
  <span
    className="main-color"
  >
    Capacity
  </span>
  <TextField
    id="capacity"
    onChange={[Function]}
    placeholder="Capacity"
    type="text"
    value=""
  />
  <span
    className="main-color"
  >
    Price
  </span>
  <TextField
    id="cost"
    onChange={[Function]}
    placeholder="NGN"
    type="text"
    value=""
  />
  <span
    className="main-color"
  >
     Description
  </span>
  <span
    className="help-block"
  />
  <p
    className="subtitle"
  >
    describe the center in few words
  </p>
  <div
    className="form-group"
  >
    <textarea
      className="form-control"
      id="description"
      onChange={[Function]}
      placeholder="Describe center in few words"
      value=""
    />
  </div>
  <input
    className="btn btn-primary basic"
    id="submit-center"
    type="submit"
    value="Add Center"
  />
</form>
`;
